hazelcast/src/main/java/com/hazelcast/sql/impl/expression/datetime/Formatter.java

```diff
@@ -0,0 +1,791 @@
+        }
+
+        static class Literal implements Part {
+            final String contents;
+
+            Literal(String contents) {
+                this.contents = contents;
+            }
+
+            @Override
+            public void format(StringBuilder s, Temporal input, Locale locale) {
+                s.append(contents);
+            }
+
+            @Override
+            public String toString() {
+                return '"' + contents + '"';
+            }
+        }
+
+        static class PatternInstance implements Part {
+            final boolean fill, translate;
+            final Pattern pattern;
+            final Ordinal ordinal;
+
+            PatternInstance(boolean fill, boolean translate, Pattern pattern, Ordinal ordinal) {
+                this.fill = fill;
+                this.translate = translate;
+                this.pattern = pattern;
+                this.ordinal = ordinal;
+            }
+
+            @Override
+            public void format(StringBuilder s, Temporal input, Locale locale) {
+                Object result = pattern.query.apply(input, translate ? locale : Locale.US);
+                String r = result.toString();
+                if (pattern == Pattern.TZH) {
+                    s.append((int) result < 0 ? '-' : '+');
+                    r = (int) result < 0 ? r.substring(1) : r;
+                }
+                if (fill) {
+                    char pad = result instanceof Number ? '0' : ' ';
+                    for (int i = r.length(); i < pattern.maxLength; i++) {
+                        s.append(pad);
+                    }
+                }
+                s.append(r);
+                if (pattern == Pattern.Y_YYY && (fill || r.length() == 4)) {
+                    s.insert(s.length() - 3, ',');
+                }
+                if (ordinal != null) {
+                    String th = getOrdinal(r);
+                    s.append(ordinal == Ordinal.TH ? th.toUpperCase() : th);
+                }
+            }
+
+            @Override
+            public String toString() {
+                return (fill ? "FM" : "") + (translate ? "TM" : "") + pattern + (ordinal == null ? "" : ordinal);
+            }
+        }
+
+        static final DateTimeFormatter
+                MERIDIEM_FORMATTER = DateTimeFormatter.ofPattern("a"),
+                TIMEZONE_FORMATTER = DateTimeFormatter.ofPattern("O"),
+                ERA_FORMATTER = DateTimeFormatter.ofPattern("G");
+
+        static final BiFunction<Object, Locale, Object>
+                UPPERCASE = (o, l) -> ((String) o).toUpperCase(l),
+                LOWERCASE = (o, l) -> ((String) o).toLowerCase(l),
+                WITH_PERIODS = (o, l) -> {
+                    String r = (String) o;
+                    return r.length() != 2 ? r : r.charAt(0) + "." + r.charAt(1) + ".";
+                };
+
+        @SuppressWarnings("checkstyle:MethodParamPad")
+        enum Pattern {
```

Comment:
We shouldn't use very common Java class names as names for our classes.

Reply:
Postgres and Oracle docs call them _template patterns_ and _format models_ respectively. JDK's choice of `Pattern` to refer regular expressions is more questionable I think since the rest of the world prefers `RegEx(p)`.